src/common/utils/collection-functions.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+/**
+ * Get the value behind `key`. If it was not pressent, first insert `value`
+ * @param map The map to interact with
+ * @param key The key to insert into the map with
+ * @param value The value to optional add to the map
+ * @returns The value in the map
+ */
+export function getOrInsert<K, V>(map: Map<K, V>, key: K, value: V): V {
+  if (map.has(key)) {
```

Comment:
Code golf:
```
if (!map.has(key)) {
    map.set(key, value);
}

return value;
```

Reply:
No those have the wrong semantics. We want to get the value in the map always. The value provided is just the default value for the key.

Reply:
Ah, thanks.